gui/pages/bundle.go

```diff
@@ -114,12 +125,84 @@ func createBundleWidget(bundle *swupd.Bundle) (*gtk.CheckButton, error) {
 	return check, nil
 }
 
+func createDecisionBox(model *model.SystemInstall, bundle *Bundle) (*decisionDialog, error) {
+	var err error
+	decisionMaker := &decisionDialog{}
+
+	decisionMaker.box, err = gtk.BoxNew(gtk.ORIENTATION_VERTICAL, 0)
+	if err != nil {
+		return nil, err
+	}
+	decisionMaker.box.SetHAlign(gtk.ALIGN_FILL)
+	decisionMaker.box.SetMarginBottom(common.TopBottomMargin)
+
+	text := utils.Locale.Get("This requires a working network connection.\nProceed with a network test?")
+	decisionMaker.label, err = common.SetLabel(text, "label-warning", 0.0)
+
+	if err != nil {
+		return nil, err
+	}
+	decisionMaker.label.SetUseMarkup(true)
+	decisionMaker.label.SetHAlign(gtk.ALIGN_START)
+	decisionMaker.box.PackStart(decisionMaker.label, false, true, 0)
+
+	decisionMaker.dialog, err = common.CreateDialogOkCancel(decisionMaker.box,
+		utils.Locale.Get("Network Required"), utils.Locale.Get("CONFIRM"), utils.Locale.Get("CANCEL"))
+
+	if err != nil {
+		return nil, err
+	}
+
+	decisionMaker.dialog.SetDeletable(false)
+
+	// Configure confirm button
+	decisionMaker.confirmButton, err = decisionMaker.dialog.GetWidgetForResponse(gtk.RESPONSE_OK)
+	if err != nil {
+		return nil, err
+	}
+
+	// Configure cancel button
+	decisionMaker.cancelButton, err = decisionMaker.dialog.GetWidgetForResponse(gtk.RESPONSE_CANCEL)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = decisionMaker.confirmButton.Connect("clicked", func() {
+		if ret, _ := network.RunNetworkTest(model); ret != network.NetTestSuccess {
+			bundle.clearPage = true
+			bundle.ResetChanges()
+			decisionMaker.dialog.Destroy()
+			return
+		}
+		bundle.clearPage = false
```

Comment:
Another force push removed the GUI code which enables the 'Confirm' button if networking passed.

```suggestion
		bundle.clearPage = false
		bundle.ResetChanges()	
```

Reply:
done